packages/flutter_tools/test/general.shard/ios/ios_project_migration_test.dart

```diff
@@ -600,6 +601,108 @@ keep this 3
         expect('Updating minimum iOS deployment target from 8.0 to 9.0'.allMatches(testLogger.statusText).length, 1);
       });
     });
+
+    group('update Xcode project object version', () {
+      late MemoryFileSystem memoryFileSystem;
+      late BufferLogger testLogger;
+      late FakeIosProject project;
+      late File xcodeProjectInfoFile;
+      late File xcodeProjectSchemeFile;
+
+      setUp(() {
+        memoryFileSystem = MemoryFileSystem();
+        testLogger = BufferLogger.test();
+        project = FakeIosProject();
+        xcodeProjectInfoFile = memoryFileSystem.file('project.pbxproj');
+        project.xcodeProjectInfoFile = xcodeProjectInfoFile;
+
+        xcodeProjectSchemeFile = memoryFileSystem.file('Runner.xcscheme');
+        project.xcodeProjectSchemeFile = xcodeProjectSchemeFile;
+      });
+
+      testWithoutContext('skipped if files are missing', () {
+        final ProjectObjectVersionMigration iosProjectMigration = ProjectObjectVersionMigration(
+          project,
+          testLogger,
+        );
+        expect(iosProjectMigration.migrate(), isTrue);
+        expect(xcodeProjectInfoFile.existsSync(), isFalse);
+        expect(xcodeProjectSchemeFile.existsSync(), isFalse);
+
+        expect(testLogger.traceText, contains('Xcode project not found, skipping Xcode compatibility migration'));
+        expect(testLogger.traceText, contains('Runner scheme not found, skipping Xcode compatibility migration'));
+        expect(testLogger.statusText, isEmpty);
+      });
+
+      testWithoutContext('skipped if nothing to upgrade', () {
+        const String xcodeProjectInfoFileContents = '''
+	objectVersion = 50;
+				LastUpgradeCheck = 1300;
```

Comment:
is this indentation right?

Reply:
You're right, will fix. It doesn't matter for the test since whitespace is ignored.

Reply:
Edit: Actually it is "right" in that it looks like
```
	classes = {
	};
	objectVersion = 50;
	objects = {
			attributes = {
				LastUpgradeCheck = 1300;
```
And I just left off the extra lines.  But I will include them since that's a better test (that no part of the file is edited)